code/modules/antagonists/traitor/objectives/eyesnatching.dm

```diff
@@ -0,0 +1,207 @@
+
+/datum/traitor_objective/eyesnatching/generate_ui_buttons(mob/user)
+	var/list/buttons = list()
+	if(!spawned_eyesnatcher)
+		buttons += add_ui_button("", "Pressing this will materialize an eyesnatcher, which can be used on incapacitaded or restrained targets to forcefully remove their eyes.", "syringe", "eyesnatcher")
+	return buttons
+
+/datum/traitor_objective/eyesnatching/ui_perform_action(mob/living/user, action)
+	. = ..()
+	switch(action)
+		if("eyesnatcher")
+			if(spawned_eyesnatcher)
+				return
+			spawned_eyesnatcher = TRUE
+			var/obj/item/eyesnatcher/eyesnatcher = new(user.drop_location())
+			user.put_in_hands(eyesnatcher)
+			eyesnatcher.balloon_alert(user, "the eyesnatcher materializes in your hand")
+
+/obj/item/eyesnatcher
+	name = "portable eyeball extractor"
+	desc = "An overly complicated device that can pierce target's skull and extract their eyeballs if enough brute force is applied."
+	icon = 'icons/obj/surgery.dmi'
+	icon_state = "eyesnatcher"
+	base_icon_state = "eyesnatcher"
+	inhand_icon_state = "hypo"
+	lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/medical_righthand.dmi'
+	throwforce = 0
+	w_class = WEIGHT_CLASS_SMALL
+	throw_speed = 3
+	throw_range = 5
+	var/used = FALSE
+
+/obj/item/eyesnatcher/update_icon_state()
+	. = ..()
+	icon_state = "[base_icon_state][used ? "-used" : ""]"
+
+/obj/item/eyesnatcher/attack(mob/living/carbon/human/victim, mob/living/user, params)
+	if(!istype(victim) || !victim.Adjacent(user)) //No TK use
+		return ..()
+
+	var/obj/item/organ/eyes/eyeballies = victim.getorgan(/obj/item/organ/eyes)
+
+	if(!eyeballies || victim.is_eyes_covered())
+		return ..()
+
+	user.do_attack_animation(victim, used_item = src)
+	victim.visible_message(span_warning("[user] presses [src] against [victim]'s skull!"), span_userdanger("[user] presses [src] against your skull!"))
+	if(!do_after(user, 5 SECONDS, target = victim, extra_checks = CALLBACK(src, .proc/eyeballs_exist, eyeballies, victim)))
+		return
+
+	to_chat(victim, span_userdanger("You feel [src] pushing at your skull!"))
+	to_chat(user, span_notice("You apply more pressure to [src]."))
+	if(!do_after(user, 5 SECONDS, target = victim, extra_checks = CALLBACK(src, .proc/eyeballs_exist, eyeballies, victim)))
+		return
+
+	var/obj/item/bodypart/head = victim.get_bodypart(BODY_ZONE_HEAD)
+	if(!head) //WUT
+		return
+
+	var/datum/wound/blunt/severe/severe_wound_type = /datum/wound/blunt/severe
+	var/datum/wound/blunt/critical/critical_wound_type = /datum/wound/blunt/critical
+	head.receive_damage(brute = 20, wound_bonus = rand(initial(severe_wound_type.threshold_minimum), initial(critical_wound_type.threshold_minimum) + 10))
+	victim.update_damage_overlays()
+	victim.audible_message(span_warning("*CRACK*"))
+	to_chat(victim, span_userdanger("[src] pierces your skull, blood flowing all over your face!"))
+	to_chat(user, span_notice("[src] pierces through [victim]'s skull."))
```

Comment:
```suggestion
	victim.visible_message(
		span_userdanger("[src] pierces your skull, blood flowing all over your face!"),
		span_danger("[src] pierces through [victim]'s skull, ripping their eyes out!"),
		span_danger("Something penetrates your skull, tearing out your eyes! Holy fuck!")
	)
```
Obviously the text changes are optional, but this proc exists specifically so you dont need to do what you did.